unity-renderer/Assets/UnityGLTF/Scripts/Serialization/Extensions/GLTFJsonExtensions.cs

```diff
@@ -81,15 +85,13 @@ public static IEnumerator ReadListDelayed<T>(this JsonReader reader, Func<T> des
                 // deserializerFunc can advance to EndArray. We need to check for this case as well. 
                 if (reader.TokenType == JsonToken.EndArray)
                     break;
-
-                throttlingCounter++;
-                if ( throttlingCounter > throttlingCounterInterval )
+                
+                if ( sw.ElapsedMilliseconds > MAX_READ_TIME_IN_MS )
                 {
                     yield return skipFrameIfDepletedTimeBudget;
-                    throttlingCounter = 0;
+                    sw.Restart();
```

Comment:
`skipFrameIfDepletedTimeBudget` is already using the throttling counter for this, so here we are having two different ways for throttling the same thing. Maybe a better approach is to just decrease the time budget for the counter? Did you try this?

Also be wary that this could increase the GLTF loading times too much.

Reply:
Hmm, you are right, I might have to do a broader test to find a sweet spot between performance impact and loading times.